### PR TITLE
Add github integration

### PR DESCRIPTION
**Pull request by mestrade**
## Push findings into github
- Enable github in the system settings
- New github configuration menu with Github profile (using api key)
- Github setup in product (choose project and Github profile)
- Support findings create/close/reopen
- Map severity with github labels

### On the code side:

- Add forms and models for github conf, product and issues
- Add github urls and views
- Add tasks for external issue in order to extract jira code later and have more abstraction

### Checks
- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [ ] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ ] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR

### CODE REVIEW

File: dojo/templates/dojo/findings_list.html [hunk 1]
```diff
@@ -69,6 +69,9 @@ <h3 class="has-filters">
                                 {% if system_settings.enable_jira %}
                                 <b>Push to JIRA</b> <input id="id_push_tojira" name="push_to_jira" type="checkbox" alt="Select to push to JIRA"/><br/>
                                 {% endif %}
+                                {% if system_settings.enable_github %}
+                                <b>Push to Github</b> <input id="id_push_togithub" name="push_to_github" type="checkbox" alt="Select to push to Github"/><br/>
```
[the remark this answers is not in the document]
Comment (mestrade) [reply dojo/templates/dojo/findings_list.html]: Labels added

File: dojo/templates/dojo/new_github.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{% extends "base.html"%}
+{% block content %}
+ {{ block.super }}
+    <h3> Add a Github Configuration </h3>
+    <form class="form-horizontal" action="{% url 'add_github' %}" method="post">{% csrf_token %}
+     {% include "dojo/form_fields.html" with form=jform %}
+        <div class="form-group">
+            <div class="col-sm-offset-2 col-sm-10">
+                <input class="btn btn-primary" type="submit" title="Express" formnovalidate=true value="Express" name="_Express"/>
+                <input class="btn btn-primary" type="submit" value="Submit"/>
```
[the remark this answers is not in the document]
Comment (mestrade) [reply dojo/templates/dojo/new_github.html]: Hmm not sure how to add a label on a submit button...
I'm not able to find labels examples in the code....

**Comment by madchap** [comment]
You need to rebase or adjust your migration file. Travis tests failing.

**Comment by mestrade** [comment]
Travis-CI randomly fail on testing :(

**Comment by madchap** [comment]
OK, this should be good to go, db migration file also updated. Merging.

**Pull request merged**